gather_istio.sh

```diff
@@ -0,0 +1,244 @@
+# Since OCP 4.16 it honors '--since' and '--since-time' args
+function inspect() {
+  local resource ns
+  resource=$1
+  ns=$2
+
+  echo
+  if [ -n "$ns" ]; then
+    echo "Inspecting resource ${resource} in namespace ${ns}"
+    # it's here just to make the linter happy (we have to use double quotes arround the variable)
+    if [ -n "${log_collection_args}" ]
+    then
+      oc adm inspect "${log_collection_args}" "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}" -n "${ns}"
+    else
+      oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}" -n "${ns}"
+    fi
+  else
+    echo "Inspecting resource ${resource}"
+    # it's here just to make the linter happy
+    if [ -n "${log_collection_args}" ]
+    then
+      oc adm inspect "${log_collection_args}" "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}"
+    else
+      oc adm inspect "--dest-dir=${BASE_COLLECTION_PATH}" "${resource}"
+    fi
+  fi
+}
+
+function inspectNamespace() {
+  local ns
+  ns=$1
+
+  inspect "ns/$ns"
+  for crd in $crds; do
+    inspect "$crd" "$ns"
+  done
+  inspect net-attach-def,roles,rolebindings "$ns"
+}
+
+function main() {
+  local crds
+  echo
+  echo "Executing Istio gather script"
+  echo
+
+  versionFile="${BASE_COLLECTION_PATH}/version"
+  echo "openshift-service-mesh/must-gather"> "$versionFile"
+  version >> "$versionFile"
+
+  # set global variable which is used when calling 'oc adm inspect'
+  get_log_collection_args
+
+  # TODO: add new name label to servicemesh-operator3 pod and use that instead of the control-plane label
+  operatorNamespace=$(oc get pods --all-namespaces -l control-plane=servicemesh-operator3 -o jsonpath="{.items[0].metadata.namespace}")
+  # this gets also logs for all pods in that namespace
+  inspect "ns/$operatorNamespace"
+  inspect clusterserviceversion "${operatorNamespace}"
+
+  inspect nodes
+
+  # TODO: we should probably add a new label or start using 'install.operator.istio.io/owning-resource'
+  # another option is to use 'metadata.ownerReferences' but having a label would be easier for a query
+  for r in $(oc get clusterroles,clusterrolebindings -l install.operator.istio.io/owning-resource -oname); do
+    inspect "$r"
+  done
+  for r in $(oc get clusterroles,clusterrolebindings -l 'app in (istiod,istio-reader)' -oname); do
+    inspect "$r"
+  done
+
+  # inspect all istio.io CRDs
+  crds="$(getCRDs)"
+  for crd in ${crds}; do
+    inspect "crd/${crd}"
+  done
+
+  # inspect all controlled mutatingwebhookconfiguration
+  for mwc in $(oc get mutatingwebhookconfiguration -l app=sidecar-injector -o name);do
+    inspect "${mwc}"
+  done
+  # inspect all controlled validatingwebhookconfiguration
+  for vwc in $(oc get validatingwebhookconfiguration -l app=istiod -o name);do
+    inspect "${vwc}"
+  done
+
+  # inspect Istio and IstioRevision resources, this will just store all CRs as those are cluster-scoped resources
+  inspect "Istio"
+  inspect "IstioRevision"
```

Comment:
Also need `IstioCNI` and `RemoteIstio` (can you add this now even though it doesn't yet exist?)

Reply:
No, the inspect would fail with "error: the server doesn't have a resource type "RemoteIstio""